connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -239,6 +253,24 @@ protected static ConfigDef baseConfigDef() {
                         Importance.LOW, HEADER_CONVERTER_CLASS_DOC);
     }
 
+    private void logDeprecationWarnings(Map<String, String> props) {
+        String[] deprecatedConfigs = new String[] {
+            INTERNAL_KEY_CONVERTER_CLASS_CONFIG,
+            INTERNAL_VALUE_CONVERTER_CLASS_CONFIG
+        };
+        for (String config : deprecatedConfigs) {
+            if (props.containsKey(config)) {
+                log.warn("Configuration {} is deprecated and will be removed in an upcoming version", config);
```

Comment:
Should we log more info about the value of the setting? Specifically, I am thinking that if we see that the configured value differs from the default value, that's particularly important, and maybe even warrants logging something at a higher log level to really get the user's attention.

Reply:
Agreed that it's worth extra emphasis if the value they've provided conflicts with the default. However--not too familiar with logging conventions in AK, so will follow your lead here, but is it really worth it to log that at the `error` level? If not, what about putting the normal deprecation messages at `info` and then upping the level to `warn` when the provided value differs from the default?

Reply:
I do wonder if it's necessary to only log the use of a deprecated configuration property (e.g., `internal.key.converter`) when the value equals the default. Or, at least have different error messages, perhaps something like:
> INFO: Configuration `internal.key.converter` is deprecated and will be removed in an upcoming release. This value matches the current default, so it can be removed without concern.

and 
> WARN: Configuration `internal.key.converter` is deprecated and will be removed in an upcoming release. The current value `my.custom.Converter` does NOT match the default of `org.apache.kafka.connect.json.JsonConverter`, and when deprecated the behavior may change.

I'm not sure how we could log at a higher level, since technically the worker could run with the configs as they currently stand.